models/yasgbt.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from tqdm.notebook import tqdm
+
+from sklearn.base import BaseEstimator
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+from sksurv.functions import StepFunction
+from sksurv.nonparametric import kaplan_meier_estimator, CensoringDistributionEstimator
+
+from models.survival_mixin import SurvivalMixin
+
+
+class DataSampler:
+    
+    def __init__(self, y):
+        self.y = y
+        times, km = kaplan_meier_estimator(y["event"], y["duration"])
+        # inverse_km has x: [0.01, ..., .99] and y: [0., ..., 820.]
+        self.inverse_km = StepFunction(x=km[::-1], y=times[::-1])
+        self.max_km, self.min_km = km[0], km[-1]
+    
+    def make_sample(self):
```

Comment:
Maybe we might want to introduce a `n_repetitions_per_iter` hyper parameter with a default to 5 or so and to call the code in `make_sample` several times before vstacking the generated training set variants to be used in a single boosting iteration.

Reply:
I'm not sure to get the intuition behind this?

Reply:
To make the fit less stochastic: the first iterations (with the largest learning rates) would be less sensitive to particular values of the sampled times.

But this would grow the memory usage by a factor of `n_repetitions_per_iter`.